api-client-util/src/main/java/com/takipi/api/client/util/cicd/ProcessQualityGates.java

```diff
@@ -0,0 +1,190 @@
+		
+		//find resurfaced Errors
+		if (resurfacedEvents) {
+			qualityReport.setResurfacedErrors(getResurfacedErrors(apiClient, input, events, deploymentStart));
+		}
+		
+		//find critical errors
+		if (input.criticalExceptionTypes != null) {
+			qualityReport.setCriticalErrors(getCriticalErrors(apiClient, input, events, deploymentStart));
+		}
+		
+		return qualityReport;
+	}
+	
+	private static List<OOReportEvent> getTopXEvents(ApiClient apiClient, RegressionInput input,
+			Collection<EventResult> events, int topIssuesVolume, DateTime deploymentStart) {
+		List<OOReportEvent> result = new ArrayList<OOReportEvent>();
+		
+		int counter = 1;
+		for (EventResult event : events) {
+			String arcLink = getArcLink(apiClient, event.id, input, deploymentStart);
+			OOReportEvent newEvent = new OOReportEvent(event, null, arcLink);
+			result.add(newEvent);
+			
+			if (counter == topIssuesVolume) {
+				break;
+			}
+			counter++;
+		}
+		return result;
+	}
+	
+	//find critical errors from error list
+	private static List<OOReportEvent> getCriticalErrors(ApiClient apiClient, RegressionInput input,
+			Collection<EventResult> events, DateTime deploymentStart) {
+		List<OOReportEvent> result = new ArrayList<OOReportEvent>();
+		for (EventResult event : events) {
+			if (!input.criticalExceptionTypes.contains(event.name)) {
+				continue;
+			}
+			String arcLink = getArcLink(apiClient, event.id, input, deploymentStart);
+			OOReportEvent newEvent = new OOReportEvent(event, null, arcLink);
+			result.add(newEvent);
+		}
+		return result;
+	}
+	
+	//find resurfaced errors from error list
+	private static List<OOReportEvent> getResurfacedErrors(ApiClient apiClient, RegressionInput input,
+			Collection<EventResult> events, DateTime deploymentStart) {
+		List<OOReportEvent> result = new ArrayList<OOReportEvent>();
+		for (EventResult event : events) {
+			if (!event.labels.contains(RegressionStringUtil.RESURFACED_ISSUE)) {
+				continue;
+			}
+			String arcLink = getArcLink(apiClient, event.id, input, deploymentStart);
+			OOReportEvent newEvent = new OOReportEvent(event, null, arcLink);
+			result.add(newEvent);
+		}
+		return result;
+	}
+	
+	//find new errors from error list
+	private static List<OOReportEvent> getNewErrors (ApiClient apiClient, RegressionInput input, Collection<EventResult> events, DateTime deploymentStart) {		
+		List<OOReportEvent> result = new ArrayList<OOReportEvent>();		
+		for (EventResult event : events) {			
+			if (!input.deployments.contains(event.introduced_by)) {
+				continue;
+			}			
+			String arcLink = getArcLink(apiClient, event.id, input, deploymentStart);
+			OOReportEvent newEvent = new OOReportEvent(event, RegressionStringUtil.NEW_ISSUE, arcLink);
+			result.add(newEvent);
+		}
+		return result;
+	}
+	
+	//get the arclink for the event
+	public static String getArcLink(ApiClient apiClient, String eventId,
+			RegressionInput input, DateTime activeWindowStart) {
+
+		DateTime activeWndowStart = activeWindowStart;
```

Comment:
Seems like `activeWndowStart` isn't needed when we got `activeWindowStart` :)

Reply:
Funny, didn't event notice. Copied from API client. Changed